utils/tfw_logger.cpp

```diff
@@ -0,0 +1,214 @@
+#include "mmap_buffer.h"
+#include "clickhouse.h"
+#include "../fw/access_log.h"
+#include <iostream>
+#include <fcntl.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <clickhouse/client.h>
+#include <clickhouse/base/socket.h>
+
+using namespace std;
+
+#define FILE_PATH	"/dev/tempesta_mmap_log"
+#define TABLE_NAME	"access_log"
+
+typedef struct {
+	const char	*name;
+	Type::Code	code;
+} TfwField;
+
+static const TfwField tfw_fields[] = {
+	[TFW_MMAP_LOG_ADDR]		= {"address", Type::IPv6},
+	[TFW_MMAP_LOG_METHOD]		= {"method", Type::UInt8},
+	[TFW_MMAP_LOG_VERSION]		= {"version", Type::UInt8},
+	[TFW_MMAP_LOG_STATUS]		= {"status", Type::UInt16},
+	[TFW_MMAP_LOG_RESP_CONT_LEN]	= {"response_content_length", Type::UInt32},
+	[TFW_MMAP_LOG_RESP_TIME]	= {"response_time", Type::UInt32},
+	[TFW_MMAP_LOG_VHOST]		= {"vhost", Type::String},
+	[TFW_MMAP_LOG_URI]		= {"uri", Type::String},
+	[TFW_MMAP_LOG_REFERER]		= {"referer", Type::String},
+	[TFW_MMAP_LOG_USER_AGENT]	= {"user_agent", Type::String},
+};
+
+TfwClickhouse *tfw_clickhouse;
```

Comment:
Apparently `tfw_clickhouse` should be a member of the same class having `callback()` as a method. Now the code is organized as the global variable, the callback using it and dynamic memory allocation for `tfw_clickhouse`. This makes the dependencies in the code vague, which may lead to spagetti, when the project grow

Reply:
> Apparently tfw_clickhouse should be a member of the same class having callback() as a method. 
Do you mean I should wrap (almost) all the code in tfw_logger.cc in a class?